Additional support for mixes of element dimensions

lindsayad opened the pull request:
- Remove flawed `isLowerD/hasLowerD` logic. The logic errors started when associating an `is_lower_d` attribute to subdomains. We (libmesh) do not require the same element type in a given subdomain, and we also do not require the same element dimension in a given subdomain. So all logic stemming from that assumed one-to-one mapping of subdomain to element dimension was also flawed.
- Add precise error checking in `ProjectionAux` where mixing dimensions is in fact logically flawed when using volume based weighting for element to node projections
- Add `GhostAllPointNeighbors` relationship manager which leverages the node-to-element maps we cache in `MooseMesh`
- Add identity weighting option when performing element to node projections

lindsayad committed:
Remove hasLowerD/isLowerD APIs predicated on subdomain arguments

This logic was fundamentally flawed as there is no requirement that
a given subdomain have all equal element dimensions

Closes #32081

lindsayad committed:
Refactor lower-d APIs in MooseMesh

lindsayad committed:
Add highest_d_blocks_only MooseVariable parameter

lindsayad committed:
Block restrict elemental aux variables to only have one elem dim value

lindsayad committed:
Revert "Block restrict elemental aux variables to only have one elem dim value"

This reverts commit 4a8ce69fe8c048eff1d9203f2de5244ba9c84ef1.

lindsayad committed:
Revert highest_d_blocks_only parameter addition

lindsayad committed:
Add devel/dbg check for single element dim in elem -> node proj

Closes #32082

lindsayad committed:
Make elem dimension check present in all METHODS

moosebuild commented:
Job [Test](https://civet.inl.gov/job/3446685/), step Results summary on 237d8c0 wanted to post the following:

## Framework test summary

Compared against 3aea247 in job [civet.inl.gov/job/3443877](https://civet.inl.gov/job/3443877).

No change
## Modules test summary

Compared against 3aea247 in job [civet.inl.gov/job/3443877](https://civet.inl.gov/job/3443877).

No change

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/3446656/), step Docs: sync website on 237d8c0 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/32068/site)

This comment will be updated on new commits.

lindsayad committed:
Move multiple boundary sides check to initialSetup

lindsayad committed:
Block restrict elemental source variable living in multiple dimensions

lindsayad committed:
Determine whether lower-d calc outside of virtual compute method

lindsayad commented on framework/src/auxkernels/AuxKernel.C:
```diff
@@ -124,7 +124,7 @@ template <typename ComputeValueType>
 AuxKernelTempl<ComputeValueType>::insert()
 {
-  if (_coincident_lower_d_calc)
+  if (_coincident_lower_d_calc.value())
```
It throws if it doesn't have a value. I guess an assertion could be used to add a more friendly error message in devel/dbg

lindsayad committed:
Make GhostHigherDLowerDPointNeighbors code match comment

lindsayad committed:
Some optimizations in GhostHigherDLowerDPointNeighbors

lindsayad committed:
Doxygen string for GhostHigherDLowerDPointNeighbors

lindsayad committed:
Create GhostAllPointNeighbors relationship manager

This is now used in ProjectionAux for enabling projections
involving a mix of element dimensions

Refs libmesh/libmesh#4352

lindsayad committed:
Add option for identity weighting for element to node projections

Closes #32083

lindsayad committed:
Address review

- Fix periodic typo
- Add comment regarding the choice of whether we are performing
  a lower-d element calculation in AuxKernel
- assert that we have made the lower-d calculation choice when
  we've made it to AuxKernel::insert()
